titan-compiler/checker.lua

```diff
@@ -367,6 +404,32 @@ check_var = function(var, errors)
     end
 end
 
+
+local function check_exp_callfunc_builtin(exp, errors)
+    local fexp = exp.exp
+    local args = exp.args
+    local builtin_name = fexp._type.builtin_decl.name
+    if builtin_name == "table.insert" then
+        check_arity("table.insert arguments", 2, #args, errors, exp.loc)
+        check_exp(args[1], errors, nil)
+        check_is_array("table.insert first argument",
```

Comment:
I think we need a little bit of meta-programming generalization here.
It won't be a blocker for this PR, but it will be necessary as others builtin functions appears.

Reply:
In this specific instance the problem would also go away if we had parametric polymorphism. I think we should see what happens with additional builtins before biting the bullet here.

Reply:
Agreed